### DISCUSSION TITLE
關於 final project 中 multiple wb request 問題

### DISCUSSION THREAD

**Discussion by samlo123** [bol-edu/HLS-SOC-Discussions#192] [Q&A]
各位大神好，
小弟這邊有一個問題想請問。

想要達成multiple wb request ，需要在一組wb 和 ack 中插入其他 wb。

這邊小弟不知道要如何達成，或是小弟有理解錯誤，希望各位大神幫幫忙。

**Comment by kevin861222** [comment]
以目前的情況來說可能沒辦法辦到，因為使用的是 Standard WB 而非 Pipeline WB，除非另外加入訊號線，你可以參考這篇文章
https://zipcpu.com/zipcpu/2017/11/07/wb-formal.html
你需要另外定義 i_wb_stall 訊號

在這篇文章中有說明到 user defined 的擴充訊號線 :
https://zeppe.pixnet.net/blog/post/28200702

（如果有回答錯誤的地方，再麻煩大神們指正，謝謝）

**Comment by boledulab** [comment]
單一 WB Bus, 是無法有 multiple requests. 但我們有 multiple WB Bus, 所以可以有 multiple outstanding requests。 有下面可能的 WB requests，
1. RISC-V CPU
2. DMA engines for accelerators, e.g. FIR, matmul, qsort
